pom.xml

```diff
@@ -55,92 +58,99 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>credentials</artifactId>
-      <version>2.1.13</version>
+      <version>2.1.18</version>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>credentials-binding</artifactId>
-      <version>1.12</version>
+      <version>1.18</version>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>authentication-tokens</artifactId>
       <version>1.3</version>
     </dependency>
+    <dependency>
+      <groupId>org.jenkinsci.plugins</groupId>
+      <artifactId>pipeline-model-extensions</artifactId>
+      <version>1.3.8</version>
+      <!-- optional because the dependency subtree is quite large -->
+      <optional>true</optional>
```

Comment:
Fair enough (for reasons similar to https://github.com/jenkinsci/aws-credentials-plugin/pull/50#discussion_r271934841), but you need to change `@Extensions` that use the optional dependency to specify `optional = true` or the plugin will fail at runtime when the optional dependency is not present. In unit/integration tests here, optional dependencies are installed, so it is easy to get this wrong, which is one of the reasons optional dependencies are painful.

Reply:
That is good to know, I marked the `@Extension` annotation on `DockerServerCredentialsHandler` as optional